packages/api/models/rules.js

```diff
@@ -206,67 +204,38 @@ class Rule extends Manager {
    */
   async addKinesisEventSource(item, lambda) {
     // use the existing event source mapping if it already exists and is enabled
-    const listParams = { FunctionName: lambda.name };
-    const listData = await aws.lambda(listParams).listEventSourceMappings().promise();
+    const listParams = {
+      FunctionName: lambda.name,
+      EventSourceArn: item.rule.value
+    };
+    const listData = await aws.lambda().listEventSourceMappings(listParams).promise();
     if (listData.EventSourceMappings && listData.EventSourceMappings.length > 0) {
-      const mappingExists = listData.EventSourceMappings
-        .find((mapping) => { // eslint-disable-line arrow-body-style
-          return (mapping.EventSourceArn === item.rule.value
-                  && mapping.FunctionArn.includes(lambda.name));
-        });
-      if (mappingExists) {
-        if (mappingExists.State === 'Enabled') {
-          return mappingExists;
-        }
-        await this.deleteKinesisEventSource({
-          name: item.name,
-          rule: {
-            arn: mappingExists.UUID,
-            type: item.rule.type
-          }
-        }, lambda.eventType);
+      const mappingExists = listData.EventSourceMappings[0];
+      if (mappingExists.State === 'Enabled') {
+        return mappingExists;
       }
+      await this.deleteKinesisEventSource({
```

Comment:
Is there a reason that we are deleting and creating instead of just updating the existing mapping?

Reply:
I don't really know. So this is an edge case really because they shouldn't be disabled anymore but it's in for backwards compatibility. I updated it to just update.